osu.Game.Rulesets.Osu/Edit/OsuSelectionHandler.cs

```diff
@@ -207,28 +192,118 @@ public override bool HandleRotation(float delta)
             return true;
         }
 
-        private bool moveSelection(Vector2 delta)
+        private bool scaleSlider(Slider slider, Vector2 scale)
+        {
+            Quad sliderQuad = getSurroundingQuad(slider.Path.ControlPoints.Select(p => p.Position.Value));
+            Vector2 pathRelativeDeltaScale = new Vector2(1 + scale.X / sliderQuad.Width, 1 + scale.Y / sliderQuad.Height);
+
+            foreach (var point in slider.Path.ControlPoints)
+                point.Position.Value *= pathRelativeDeltaScale;
+
+            //if sliderhead or sliderend end up outside playfield, revert scaling.
+            Quad scaledQuad = getSurroundingQuad(new OsuHitObject[] { slider });
+            (bool xInBounds, bool yInBounds) = isQuadInBounds(scaledQuad);
+
+            if (xInBounds && yInBounds)
+                return true;
+
+            foreach (var point in slider.Path.ControlPoints)
+                point.Position.Value *= new Vector2(1 / pathRelativeDeltaScale.X, 1 / pathRelativeDeltaScale.Y);
+
+            return true;
```

Comment:
Both return points of this method (and the only one in `scaleHitObjects()`, as well) specify `true` as the return values. Is this correct? Seems a bit pointless to return anything if so.

Also this method of reverting (doing `* scale`, but then `* (1 / scale)`) may be mathematically sound, but will not go well with floating point arithmetic (may not revert perfectly). Probably best to take the hit and copy the entire slider path out (or at least the control point coords) and store them back if a revert is required.

Reply:
I have removed the return values for now and left `HandleScale` always returning true like the other `HandleX` methods in this class. I couldn't find any point where the return value is used.

Good point about the reverting.